utils/monitor.js

```diff
@@ -0,0 +1,81 @@
+const { NotFound } = require("http-errors");
+const fireStore = require("./firestore");
+const trackedProgressesCollection = fireStore.collection("trackedProgresses");
+const { RESPONSE_MESSAGES } = require("../constants/monitor");
+const { RESOURCE_NOT_FOUND } = RESPONSE_MESSAGES;
+
+/**
+ * Builds a Firestore query to check if a document exists based on the provided query parameters.
+ *
+ * @param {Object} queryParams - The query parameters for checking if a document exists.
+ * @param {string} queryParams.userId - The ID of the user (optional).
+ * @param {string} queryParams.taskId - The ID of the task (optional).
+ * @returns {Firestore.Query} - A Firestore query to check if a document exists.
+ */
+const buildQueryToCheckIfDocExists = (queryParams) => {
+  const { userId, taskId } = queryParams;
+  if (userId) {
+    return trackedProgressesCollection.where("userId", "==", userId);
+  } else {
+    return trackedProgressesCollection.where("taskId", "==", taskId);
+  }
+};
+
+/**
+ * Builds a Firestore query for fetching tracked progress documents of a specific type based on the provided query parameters.
+ *
+ * @param {Object} queryParams - The query parameters for fetching tracked progress documents.
+ * @param {string} queryParams.type - The type of tracked progress documents.
+ * @param {boolean} queryParams.monitored - The flag indicating if the documents are marked (optional).
+ * @returns {Firestore.Query} - A Firestore query for fetching tracked progress documents of a specific type.
+ */
+const buildQueryForFetchingDocsOfType = (queryParams) => {
+  const { type, monitored } = queryParams;
+  let query = trackedProgressesCollection.where("type", "==", type);
+  if (monitored !== undefined) {
+    query = query.where("monitored", "==", JSON.parse(monitored));
+  }
```

Comment:
Since this generic query can be reused at multiple places it was suggested by few of the reviewers in few of my earlier PRs to abstract in a util function. https://github.com/Real-Dev-Squad/website-backend/pull/1074#discussion_r1190106864. Let me know your thoughts. In the current code as well few of the queries have been used at multiple places.